src/app/leaves/components/LeaveActionRender.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo } from "react";
+import ActionsMenu from "@/common/ActionMenu/ActionsMenu";
+import {
+  Info as InfoIcon,
+  Trash2 as TrashIcon,
+  Edit as EditIcon,
+  CheckCircle as CheckCircleIcon,
+  AlertCircle as AlertCircleIcon,
+} from "lucide-react";
+import { MemberRole } from "@/lib/constants";
+
+const loggedUser = {
+  name: "John Smith",
+  role: "member",
+};
+
+const LeaveTableActionRender = ({
+  leaveRequestStatus,
+}: {
+  leaveRequestStatus: string | undefined;
+}) => {
+  const handleViewDetails = () => {
+    console.log("Viewing details...");
+  };
+
+  const handleEditInfo = () => {
+    console.log("Editing info...");
+  };
+
+  const handleDeleteRequest = () => {
+    console.log("Deleting request...");
+  };
+
+  const handleApprovedRequest = () => {
+    console.log("Approved Request...");
+  };
+
+  const handleRejectedRequest = () => {
+    console.log("Rejected Request...");
+  };
+
+  const baseActions = [
+    {
+      title: "Edit",
+      id: 2,
+      onClick: handleEditInfo,
+      icon: <EditIcon size={16} />,
+    },
+    {
+      title: "Delete",
+      id: 3,
+      onClick: handleDeleteRequest,
+      icon: <TrashIcon size={16} />,
+    },
+  ];
+
+  const viewInfo = [
```

Comment:
We're creating bunch of arrays but none of them is memoized.

Reply:
I have memoized all arrays